tests/foreman/installer/test_installer.py

```diff
@@ -1808,3 +1813,32 @@ def test_satellite_installation(installer_satellite):
     assert installer_satellite.execute('rpm -q foreman-redis').status == 0
     settings_file = installer_satellite.load_remote_yaml_file(FOREMAN_SETTINGS_YML)
     assert settings_file.rails_cache_store.type == 'redis'
+
+
+@pytest.mark.parametrize('package', ['nmap-ncat'])
+def test_weak_dependency(sat_non_default_install, package):
+    """Check if Satellite and its (sub)components do not require certain (potentially insecure) packages
+
+    :id: c7988920-2f8c-4646-bde9-8823a3ca96bb
+
+    :steps:
+        1. Use satellite with non-default setup (for 'nmap-ncat' enable foreman discovery plugin and install foreman-discovery-image)
+        2. Attempt to remove the package
+
+    :expectedresults:
+        1. The package can be either not installed or can be removed without removing any Satellite or Foreman packages
```

Comment:
Running the test for `nmap-ncat` package makes sense only if the package was installed.
What benefit do you see in running the test if  `nmap-ncat` is not installed on the system?

Note: If you decide to modify the `expectedresults` field, the last assertion also needs to be updated.

Reply:
+1
You cannot test in a single test both that rpm is not present and that rpm gets removed.
This test is targeted to test that weak dependencies stay weak by removal transaction.

So please lets not mix things together, there should be assert only about removal so that we  will know in future FDI no longer requires nmap (won't get installed anymore) 

With broad `OR` assert we won't never get to know.

Reply:
I've change my mind after discussion with @rmynar 
The test not only tests weak dependency but also no dependency - we do not care if it is `no` OR `weak` dependency as long as it is so we are good.